plugins/AbstractEddPlugin.php

```diff
@@ -0,0 +1,51 @@
+<?php
+/**
+ * Abstract Easy Digital Downloads (EDD) Plugin.
+ *
+ * @package Junaidbhura\Composer\WPProPlugins\Plugins
+ */
+
+namespace Junaidbhura\Composer\WPProPlugins\Plugins;
+
+use Composer\Semver\Semver;
+use UnexpectedValueException;
+
+/**
+ * AbstractEddPlugin class.
+ */
+abstract class AbstractEddPlugin extends AbstractPlugin {
+
+	/**
+	 * Get the download URL for this plugin.
+	 *
+	 * @param  array<string, mixed> $response The EDD API response.
+	 * @return string
+	 */
+	protected function extractDownloadUrl( array $response ) {
+		if ( empty( $response['download_link'] ) || ! is_string( $response['download_link'] ) ) {
+			throw new UnexpectedValueException(sprintf(
```

Comment:
@mcaskill Can we add spaces in here and everywhere in this file to be consistent with WP Coding standards?